app/lib/features/space/sheets/link_chat_sheet.dart

```diff
@@ -0,0 +1,224 @@
+          children: [
+            parentSpaceSelector(context, ref),
+            Search(
+              onChanged: (value) => ref
+                  .read(chatSearchValueProvider.notifier)
+                  .update((state) => value),
+              searchController: searchTextEditingController,
+            ),
+            Expanded(child: chatsList(context, ref)),
+          ],
+        ),
+      ),
+    );
+  }
+
+  //Parent space selector
+  Widget parentSpaceSelector(BuildContext context, WidgetRef ref) {
+    return const Padding(
+      padding: EdgeInsets.all(16),
+      child: SelectSpaceFormField(
+        canCheck: 'CanLinkSpaces',
+        mandatory: true,
+        title: 'Parent space',
+        emptyText: 'optional parent space',
+        selectTitle: 'Select parent space',
+      ),
+    );
+  }
+
+  //List of chats excluding DMs that can be linked according to the selected parent space
+  Widget chatsList(BuildContext context, WidgetRef ref) {
+    final searchValue = ref.watch(chatSearchValueProvider);
+    if (searchValue != null && searchValue.isNotEmpty) {
+      return ref.watch(searchedChatsProvider).when(
+            data: (chats) {
+              var chatList =
+                  chats.where((element) => (!element.isDm())).toList();
+              if (chatList.isEmpty) {
+                return const Center(
+                  heightFactor: 10,
+                  child: Text('No chats found matching your search term'),
+                );
+              }
+              return ListView.builder(
+                padding: const EdgeInsets.all(8),
+                itemCount: chatList.length,
+                itemBuilder: (context, index) {
+                  return chatsListItemUI(chatList[index].getRoomIdStr());
+                },
+              );
+            },
+            loading: () => const Center(
+              heightFactor: 10,
+              child: CircularProgressIndicator(),
+            ),
+            error: (e, s) => Center(
+              heightFactor: 10,
+              child: Text(
+                'Searching failed: $e',
+              ),
+            ),
+          );
+    }
+
+    final chats =
+        ref.watch(chatsProvider).where((element) => (!element.isDm())).toList();
+    return ListView.builder(
+      padding: const EdgeInsets.all(8),
+      itemCount: chats.length,
+      itemBuilder: (context, index) {
+        return chatsListItemUI(chats[index].getRoomIdStr());
+      },
+    );
+  }
+
+  //Chats list item UI
+  Widget chatsListItemUI(String roomId) {
+    final roomItem = ref.watch(briefRoomItemWithMembershipProvider(roomId));
+
+    return roomItem.when(
+      data: (roomProfile) {
+        final membership = roomProfile.membership;
+        final canLink =
+            membership == null ? false : membership.canString('CanLinkSpaces');
+        final isLinked = subChatsIds.contains(roomId);
+
+        return widget.parentSpaceId == roomId
```

Comment:
Similarly I'd expect all this in the same providers as well. Like as little of this as possible should be happening in the widget and rather in the providers...